keps/sig-storage/3476-volume-group/README.md

```diff
@@ -0,0 +1,1647 @@
+
+```
+  rpc ModifyVolume(ModifyVolumeRequest)
+    returns (ModifyVolumeResponse) {
+        option (alpha_method) = true;
+    }
+```
+
+This RPC is called when an existing volume is added to an existing volume group or when a volume is removed from the volume group.
+A volume group id parameter will be in the ModifyVolumeRequest for an add request.
+A volume group id parameter will not be in the ModifyVolumeRequest for a delete request.
+If user requests to add an existing volume to a consistency group, but the CSI driver cannot fulfill the request because the existing volume is placed on a different storage pool from the consistency group, then the CSI driver MUST return failure.
+This RPC MUST be idempotent.
+
+```
+message ModifyVolumeRequest {
+      string volume_id = 1;
+
+      // This field is OPTIONAL.
+      repeated string volume_group_id = 2 [(alpha_field) = true];
+
+      // Secrets required by plugin to complete modify volume request.
+      // This field is OPTIONAL. Refer to the `Secrets Requirements`
+      // section on how to use this field.
+      map<string, string> secrets = 3 [(csi_secret) = true];
+}
+```
+External-provisioner will be modified so that modifying PVC by adding VolumeGroupName will trigger a ModifyVolume call (a new CSI controller RPC) to CSI driver.
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Other
+    - Describe the mechanism:
+      The external volume group and group snapshot controllers do not have a
+      feature gate because they are out of tree.
+      It is enabled when these external controller sidecars are deployed with the CSI driver.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      From the controller side, it only affects the external controller sidecars.
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      No.
+
+* **Does enabling the feature change any default behavior?**
+  Yes. Enabling the feature can allow a new PVC to be created and added to a VolumeGroup. Enabling the feature can also allow a VolumeSnapshot to be created as part of the VolumeSnapshotGroup.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes. All VolumeGroup and VolumeGroupSnapshot API objects need to be deleted before this feature can be truly disabled.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  We will be able to create new VolumeGroup and VolumeGroupSnapshot API objects again.
+
+* **Are there any tests for feature enablement/disablement?**
+  Unit tests will be added for the in-tree feature enable/disablement.
```

Comment:
Should this line be removed or is there an in-tree enablement aspect of this? It seems like this is entirely out of tree and the next sentence mentions there won't be tests for feature enablement/disablement.

Reply:
Nice catch.  I'll fix it.